bin/pre-commit-hook.js

```diff
@@ -37,22 +37,30 @@ function phpcsFilesToFilter( file ) {
 /**
  * Provides filter to determine which JS files to run through Prettify and linting.
  *
- * @param {String} file File name of js file modified.
- * @return {boolean}        If the file matches the requirelist.
+ * @param {string} file - File name of js file modified.
+ * @returns {boolean} If the file matches the requirelist.
  */
 function filterJsFiles( file ) {
 	return [ '.js', '.json', '.jsx' ].some( extension => file.endsWith( extension ) );
 }
 
-// Filter callback for JS files
+/**
+ * Filter callback for JS files
+ *
+ * @param {string} file - dirty file
+ * @returns {boolean} whether file needs to be linted
+ */
 function filterEslintFiles( file ) {
 	const rootMatch = /^([a-zA-Z-]+\.)/g; // *.js(x)
-	const folderMatches =
-		file.startsWith( '_inc' ) || file.startsWith( 'extensions' ) || file.startsWith( 'modules' );
+	console.log( '!!!!!!', file );
+	const folderArray = [ '_inc', 'extensions', 'modules', 'tests/e2e' ];
+	const folderMatches = folderArray.some( folder => file.includes( folder ) );
```

Comment:
I note that switching from `startsWith` to `includes` here might make it start linting additional files.

Currently the only one that seems like it'll be added is tests/jest-extensions-babel-transform.js.

Reply:
I wanted to make that list easily extendable, so I went for `includes` approach. Happy to revert it back to the original approach if you think it makes sense.

Reply:
Seems easily extendible either way. The only difference is whether the list contains file path prefixes (`startsWith`) or substrings that may occur anywhere in the file path (`includes`).

That's why tests/jest-extensions-babel-transform.js is now being linted, it happens to contain "extensions" in the middle of the filename. If that's intended, fine, but it doesn't seem intended since the variables are named `folderArray` and `folderMatches`.

Reply:
I parsed your initial comment completely wrong 🤦

So you fine with the usage of `folderArray.some()` instead of multiple `||`, and you think that we should still use `.startsWith()` in the callback.   

I switched back to `startsWith()` in b466fd0